fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergAlterTableExecutor.java

```diff
@@ -610,31 +610,30 @@ private void scanAndDeleteInvalidFiles(String tableLocation, long expiration, Se
                 if (status.getModificationTime() < expiration && !validFiles.contains(entry.getPath().getName())) {
                     filesToDelete.add(entry.getPath());
                     if (filesToDelete.size() >= DELETE_BATCH_SIZE) {
-                        filesToDelete.forEach(path -> {
-                            try {
-                                fileSystem.delete(path, false);
-                                LOGGER.debug("Deleted file {}", path);
-                            } catch (IOException e) {
-                                LOGGER.error("Failed to delete file {}", path, e);
-                            }
-                        });
+                        deleteFiles(fileSystem, filesToDelete);
                         filesToDelete.clear();
                     }
                 }
             }
             if (!filesToDelete.isEmpty()) {
-                filesToDelete.forEach(path -> {
-                    try {
-                        fileSystem.delete(path, false);
-                        LOGGER.debug("Deleted file {}", path);
-                    } catch (IOException e) {
-                        LOGGER.error("Failed to delete file {}", path, e);
-                    }
-                });
+                deleteFiles(fileSystem, filesToDelete);
                 filesToDelete.clear();
             }
         } catch (IOException e) {
             throw new StarRocksConnectorException("Failed accessing data: ", e);
         }
     }
+
+    // TODO:implement deleteFiles in FsUtils
```

Comment:
I believe Starrocks needs to implement its upper FileSystem class like Trino did.